src/js/containers/aboutTheData/modals/UnlinkedDataContainer.jsx

```diff
@@ -0,0 +1,78 @@
+/**
+ * UnlinkedDataContainer.jsx
+ * Created by Jonathan Hill 01/15/21
+ */
+
+import React, { useState, useEffect, useRef } from 'react';
+import PropTypes from 'prop-types';
+import { Table } from 'data-transparency-ui';
+import { isCancel } from 'axios';
+import { unlinkedDataColumns } from 'dataMapping/aboutTheData/modals';
+import { formatUnlinkedDataRows, fetchMockUnlinkedData } from 'helpers/aboutTheDataHelper';
+
+const propTypes = {
+    agencyData: PropTypes.shape({
+        agencyName: PropTypes.string,
+        agencyCode: PropTypes.string,
+        fiscalYear: PropTypes.number,
+        fiscalPeriod: PropTypes.number,
+        type: PropTypes.string
+    })
+};
+
+const UnlinkedDataContainer = ({ agencyData }) => {
+    const [loading, setLoading] = useState(true);
+    const [error, setError] = useState({ error: false, message: '' });
+    const [rows, setRows] = useState([]);
+    const unlinkedDataReq = useRef(null);
+
+    const unlinkedDataRequest = async () => {
+        if (error.error) setError({ error: false, message: '' });
+        if (!loading) setLoading(true);
+        if (unlinkedDataReq.current) unlinkedDataReq.current.cancel();
+        try {
+            unlinkedDataReq.current = fetchMockUnlinkedData(
+                agencyData.agencyCode,
+                agencyData.fiscalYear,
+                agencyData.fiscalPeriod,
+                agencyData.type === 'Contract' ? 'procurement' : 'assistance'
+            );
+            const res = await unlinkedDataReq.current.promise;
+            setRows(formatUnlinkedDataRows(res, agencyData.type));
+            setLoading(false);
+            unlinkedDataReq.current = null;
+        }
+        catch (e) {
+            if (!isCancel(e)) {
+                setLoading(false);
+                setError({ error: true, message: e.message });
+                console.error(e);
+            }
+            unlinkedDataReq.current = null;
+        }
+    };
+    useEffect(() => {
+        unlinkedDataRequest();
+        return () => {
+            if (unlinkedDataReq.current) unlinkedDataReq.current.cancel();
+        };
+    }, []);
+    const columns = unlinkedDataColumns(agencyData.type).map((column, i) => ({
```

Comment:
could be defined outside component to avoid having to define this function again everytime we invoke this component.

Reply:
Moved the function to the component to get rid of the variable but we need to update the columns based on contracts or financial assistance. Maybe we could move this to a useEffect hook and set a property in state?